### PR TITLE
FlxNapeSprite: add support for compound bodies

### PR DESCRIPTION
**Pull request by thegoodideaco**
If the body is part of a compound, then it's space cannot be changed, instead must be added/removed from the compound bodies list

### CODE REVIEW

File: flixel/addons/nape/FlxNapeSprite.hx [hunk 1]
```diff
@@ -95,6 +95,7 @@ class FlxNapeSprite extends FlxSprite
 	}
 
 	/**
+	/**
```
Comment (Gama11) [inline flixel/addons/nape/FlxNapeSprite.hx]: Seems the `/**` here was duplicated?

**Comment by Gama11** [comment]
Any nape demo or code I could use to test this with?

**Comment by thegoodideaco** [comment]
I can make a quick example if you like.

File: flixel/addons/nape/FlxNapeSprite.hx (hunk 1 above)
Comment (thegoodideaco) [inline flixel/addons/nape/FlxNapeSprite.hx]: ahh crap, fixing

File: flixel/addons/nape/FlxNapeSprite.hx [hunk 2]
```diff
@@ -116,10 +121,14 @@ class FlxNapeSprite extends FlxSprite
 	{
 		super.revive();
 		
-		if (body != null)
+		if (body != null && body.compound == null)
```
Comment (Gama11) [inline flixel/addons/nape/FlxNapeSprite.hx]: Are you sure it shouldn't look like this?

```haxe
if (body != null)
{
	if (body.compound == null)
		body.space = FlxNapeSpace.space;
	else
		body.compound.bodies.add(body);
}
````

Right now it would add `null` to `compound.bodies`.

**Comment by thegoodideaco** [comment]
@Gama11 : https://github.com/thegoodideaco/FlxNapeSpriteTest for an example.

**Comment by Gama11** [comment]
Thanks!

`set_physicsEnabled()` can still cause a crash. Should probably move the logic for adding a body / removing a body into separate methods to avoid redundancy and to be usable from there.

**Comment by Gama11** [comment]
Any update? :)

**Comment by thegoodideaco** [comment]
Should be ready to rock

File: flixel/addons/nape/FlxNapeSprite.hx [hunk 3]
```diff
@@ -217,8 +226,23 @@ class FlxNapeSprite extends FlxSprite
...
 		{
-			if (FlxNapeSpace.space != null)
-				FlxNapeSpace.space.bodies.remove(body);
+			
+			body.constraints.foreach(function(c:Constraint)
+			{
+				c.active = false;
+			});
+			
+			if (body.compound == null)
```
Comment (Gama11) [inline flixel/addons/nape/FlxNapeSprite.hx]: This code section looks like an exact copy of the one in `revive()`.

File: flixel/addons/nape/FlxNapeSprite.hx [hunk 4]
```diff
@@ -217,8 +226,23 @@ class FlxNapeSprite extends FlxSprite
 	{
 		if (body != null) 
 		{
-			if (FlxNapeSpace.space != null)
-				FlxNapeSpace.space.bodies.remove(body);
+			
+			body.constraints.foreach(function(c:Constraint)
```
Comment (Gama11) [inline flixel/addons/nape/FlxNapeSprite.hx]: Why is this necessary?

**Comment by Gama11** [comment]
The most recent commit doesn't seem to have any impact on the `set_physEnabled` crash:

```
Error: Cannot set the space of a Body belonging to a Compound, only the root Compound space can be set
```

(see https://github.com/Gama11/FlxNapeSpriteTest)

**Comment by thegoodideaco** [comment]
This is odd, let me look

**Comment by thegoodideaco** [comment]
i think we need to add an additional property to store any compound it might be apart of. If we disable physics, we have to remove it from the compound instead of from space. We need something to reference if they enable physics again

**Comment by Gama11** [comment]
Hm... this is getting pretty complex. I should finally get around to setting up unit tests for flixel-addons so we can test all these different cases..

**Comment by thegoodideaco** [comment]
I agree

**Comment by Gama11** [comment]
Closing this for now since there's still some unresolved issues, I created an issue as a reminder: #321
